pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -327,21 +326,12 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
                 LightningDataModule.__class__.__name__: pl DataModule's state
             }
         """
-
-        # dump epoch/global_step/pytorch-lightning_version
-        current_epoch = self.trainer.current_epoch
-        global_step = self.trainer.global_step
-        has_reached_max_steps = _is_max_limit_reached(global_step, self.trainer.max_steps)
-
-        global_step += 1
-        if not has_reached_max_steps:
-            current_epoch += 1
-
         model = self.trainer.lightning_module
 
         checkpoint = {
-            "epoch": current_epoch,
-            "global_step": global_step,
+            # the epoch is saved for compatibility but it's not relevant for restoration
+            "epoch": self.trainer.current_epoch,
+            "global_step": self.trainer.global_step + 1,
```

Comment:
Should we increment global_step there? Normally, it shouldn't be if a failure was raised. Am I missing something ?

Reply:
That's how it works in master, it gets increased unconditionally:

https://github.com/PyTorchLightning/pytorch-lightning/blob/5e78f429e3a79ac2c7c7402f922c179775db88a4/pytorch_lightning/trainer/connectors/checkpoint_connector.py#L338

Preparing a fix (after this PR) in #11805